citesphere/src/main/java/edu/asu/diging/citesphere/api/v1/user/JobInfoController.java

```diff
@@ -54,24 +57,26 @@ public ResponseEntity<String> getProfile(@RequestHeader HttpHeaders headers) {
         IJob job = jobManager.findJob(tokenContents.getJobId());
         IZoteroToken zoteroToken = tokenManager.getToken(userManager.findByUsername(job.getUsername()));
         
-        ObjectMapper mapper = new ObjectMapper();
-        ObjectNode node = mapper.createObjectNode();
-        node.put("zotero", zoteroToken.getToken());
-        node.put("zoteroId", zoteroToken.getUserId());
-        node.put("username", job.getUsername());
+        Map<String, Object> reponse = new HashMap<>();
+        reponse.put("zotero", zoteroToken.getToken());
+        reponse.put("zoteroId", zoteroToken.getUserId());
+        reponse.put("username", job.getUsername());
         // FIXME: ugly, needs better solution
         if (job instanceof IUploadJob) {
-            node.put("groupId", ((IUploadJob)job).getCitationGroup());
-        }
-        if (job instanceof IExportJob) {
+            reponse.put("groupId", ((IUploadJob)job).getCitationGroup());
+        } else if (job instanceof IExportJob) {
             IExportTask exportTask = exportTaskManager.get(((IExportJob)job).getTaskId());
-            node.put("groupId", exportTask.getGroupId());
-            node.put("collectionId", exportTask.getCollectionId());
-            node.put("exportType", exportTask.getExportType().name());
-            node.put("taskId", exportTask.getId());
+            reponse.put("groupId", exportTask.getGroupId());
+            reponse.put("collectionId", exportTask.getCollectionId());
+            reponse.put("exportType", exportTask.getExportType().name());
+            reponse.put("taskId", exportTask.getId());
+        } else if (job instanceof IImportCrossrefJob) {
+            List<String> doisList = new ArrayList<>();
+            ((IImportCrossrefJob)job).getDois().forEach(d -> doisList.add(d));
+            reponse.put("dois", doisList);
+            reponse.put("groupId", ((IImportCrossrefJob)job).getCitationGroup());
```

Comment:
Not sure where the comment went, but I made one here before about handling this differently (put the code in methods and map those to job types).